plugins/mimecast/komand_mimecast/tasks/monitor_siem_logs/task.py

```diff
@@ -114,6 +138,24 @@ def _filter_and_sort_recent_events(
             ),
             key=itemgetter(EventLogs.FILTER_DATETIME),
         )
+
+        if len(task_output) > MAX_EVENTS_PER_RUN:
+            start_point = last_log_line
+            end_point = start_point + MAX_EVENTS_PER_RUN
+
+            if end_point > len(task_output) - 1:
+                end_point = len(task_output) - 1
+                last_log_line = 0
+            else:
+                last_log_line = end_point
+
+            self.logger.info(
+                f"Number of returned logs after filtering performed was greater than {MAX_EVENTS_PER_RUN}, limiting to {MAX_EVENTS_PER_RUN}.\n"
+                f"fetching logs from {start_point} to {end_point} with a filter time of {filter_time}"
+            )
+
+            task_output = task_output[start_point:end_point]
+
         latest_time = filter_time
         for event in task_output:
```

Comment:
I think we can remove this, we already iterate over all the logs to compare datetime so we know the events are sorted. We should be able to grab the last index of the list to get the latest time, and afaik any fields not on the parser will just be dropped. So won't cause any issues downstream either (we should double check this)

Reply:
As discussed with IDR if the user selects to parse raw logs then they will still this field in the logs.

A new ticket can be raised around better handling of this logic to avoid the need to not re-loop over the list